### PR TITLE
app-forensics/aflplusplus: add 4.01c

### PR DESCRIPTION
**Pull request by miller-alex**
Version bump to app-forensics/aflplusplus-4.01c. Works for me with clang-14.

The first commit is the bare minimum with new ebuild and fixed patches, other improvements as separate commits so you can pick what you like. You may squash them after review if you prefer.

What still needs to be done:
* test with python3_11 and enable it if it works,
* decide what to do with llvm/clang-15 support (after release? lld-15 issues?),
* there are still lines in the makefiles ignoring errors that probably shouldn't (needs investigation, no priority),
* probably more things I missed.

This took a bit longer than expected because I messed up and realized only much later. Sorry for the delay.

### CODE REVIEW

**Review by thesamesam — COMMENTED** [review]
Looks great! Not tried it yet but I'm sure it's good.

Some minor comments below.

File: app-forensics/aflplusplus/files/aflplusplus-4.01c-lld-detect.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--- a/GNUmakefile.llvm
```
[the remark this answers is not in the document]
Comment (miller-alex) [reply app-forensics/aflplusplus/files/aflplusplus-4.01c-lld-detect.patch:1]: From the build log:
`
make -f GNUmakefile.llvm
make[1]: Entering directory '/tmp/portage-alex/portage/app-forensics/aflplusplus-4.01c/work/AFLplusplus-4.01c'
[+] llvm_mode detected llvm 10+, enabling neverZero implementation and c++14
[+] llvm_mode detected llvm 11+, enabling afl-lto LTO implementation
GNUmakefile.llvm:224: ld.lld not found, cannot enable LTO mode
`
Don't remember what exactly that means, but chose to include the patch to silence the warning.

[the remark this answers is not in the document]
Comment (miller-alex) [reply app-forensics/aflplusplus/files/aflplusplus-4.01c-lld-detect.patch:1]: Just checked again and a bunch of symlinks to the executable and some files in /usr/lib64/afl/ (afl-llvm-rt-lto*.o, SanitizerCoverageLTO.so, afl-llvm-lto-instrumentlist.so) are missing without the patch.

[the remark this answers is not in the document]
Comment (thesamesam) [reply app-forensics/aflplusplus/files/aflplusplus-4.01c-lld-detect.patch:1]: Thanks!

**Comment by thesamesam** [comment]
Big thanks. Merged! (Lost Closes tag for PR w/ squashing fixup).

You're more than welcome to join in co-maintenance if interested.

**Pull request closed without merging**